llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -753,121 +709,113 @@ PreservedAnalyses SpecConstantsPass::run(Module &M,
+            auto *Int32Ty = Type::getInt32Ty(Ctx);
+            SmallVector<Metadata *, 16> MDOps;
+
+            // Copy the existing metadata.
+            MDOps.push_back(Prev.second->getOperand(0));
+            MDOps.push_back(Prev.second->getOperand(1));
+            MDOps.push_back(Prev.second->getOperand(2));
+            auto &SizeOp = Prev.second->getOperand(3);
+            MDOps.push_back(SizeOp);
+
+            // Extract the size of the previous node to use as CompositeOffset
+            // for the padding node.
+            auto PrevSize = mdconst::extract<ConstantInt>(SizeOp)->getValue();
+
+            // The max value is a magic value used for padding that the
+            // runtime knows to skip.
+            MDOps.push_back(ConstantAsMetadata::get(Constant::getIntegerValue(
+                Int32Ty, APInt(32, std::numeric_limits<unsigned>::max()))));
+            MDOps.push_back(ConstantAsMetadata::get(
+                Constant::getIntegerValue(Int32Ty, PrevSize)));
+            MDOps.push_back(ConstantAsMetadata::get(
+                Constant::getIntegerValue(Int32Ty, APInt(32, Padding))));
+
+            // Replace previous metadata node with the node including the
+            // padding.
+            SCMetadata[Prev.first] = MDNode::get(Ctx, MDOps);
           }
 
-          Type *Int8Ty = Type::getInt8Ty(CI->getContext());
-          Type *Int32Ty = Type::getInt32Ty(CI->getContext());
-          GetElementPtrInst *GEP = GetElementPtrInst::Create(
-              Int8Ty, RTBuffer,
-              {ConstantInt::get(Int32Ty, CurrentOffset, false)}, "gep", CI);
-
-          Instruction *BitCast = nullptr;
-          if (SCTy->isIntegerTy(1)) // No bitcast to i1 before load
-            BitCast = GEP;
-          else
-            BitCast = new BitCastInst(
-                GEP, PointerType::get(SCTy, GEP->getAddressSpace()), "bc", CI);
-
-          // When we encounter i1 spec constant, we still load the whole byte
-          Replacement = new LoadInst(SCTy->isIntegerTy(1) ? Int8Ty : SCTy,
-                                     BitCast, "load", CI);
-          if (SCTy->isIntegerTy(1)) // trunc back to i1 if necessary
-            Replacement = CastInst::CreateIntegerCast(
-                Replacement, SCTy, /* IsSigned */ false, "tobool", CI);
-        } else {
-          // Replace the intrinsic with default C++ value for the spec constant
-          // type.
-          Replacement = getDefaultCPPValue(SCTy);
+          SCMetadata[SymID] = generateSpecConstantMetadata(
+              M, SymID, SCTy, NextID, /* is native spec constant */ false);
+
+          ++NextID.ID;
+          NextOffset += Size;
         }
+
+        Type *Int8Ty = Type::getInt8Ty(CI->getContext());
+        Type *Int32Ty = Type::getInt32Ty(CI->getContext());
+        GetElementPtrInst *GEP = GetElementPtrInst::Create(
+            Int8Ty, RTBuffer,
+            {ConstantInt::get(Int32Ty, CurrentOffset, false)}, "gep", CI);
```

Comment:
I understand, that this code didn't change, but still want to suggest: it seem to be a good practice in LLVM passes to name IR variables adding the pass name to it. So what about 'gep' -> 'gep.spec.const' or something like this?